neutrino.go

```diff
@@ -945,13 +946,19 @@ cleanup:
 	log.Tracef("Peer handler done")
 }
 
-// addrStringToNetAddr takes an address in the form of 'host:port' and returns
-// a net.Addr which maps to the original address with any host names resolved
-// to IP addresses.
-func addrStringToNetAddr(addr string) (net.Addr, error) {
+// addrStringToNetAddr takes an address in the form of 'host:port' or 'host'
+// and returns a net.Addr which maps to the original address with any host
+// names resolved to IP addresses and a default port added, if not specified,
+// from the ChainService's network parameters.
+func (s *ChainService) addrStringToNetAddr(addr string) (net.Addr, error) {
 	host, strPort, err := net.SplitHostPort(addr)
 	if err != nil {
-		return nil, err
+		if strings.HasSuffix(err.Error(), "missing port in address") {
```

Comment:
Rather than examining the string, I think we can check directly against the [`AddrError`](https://golang.org/pkg/net/#AddrError) type.

Reply:
I've tested this; it works and I've updated the code. Thanks!